File: 3567-convert-date-to-binary/convert-date-to-binary.py
class Solution:
    def convertDateToBinary(self, date: str) -> str:
        l = date.split("-")
        s = ""
        for i in l:
            s = s + str(bin(int(i))[2:]) + '-'
        s = s.rstrip(s[-1])
        return s

File: 3567-convert-date-to-binary/convert-date-to-binary.java
class Solution {
    public String convertDateToBinary(String date) {
        String[] l = date.split("-");
        StringBuilder s = new StringBuilder();
        for (String i : l) {
            s.append(Integer.toBinaryString(Integer.parseInt(i))).append('-');
        }
        s.deleteCharAt(s.length() - 1);
        return s.toString();
    }
}
